addons/website/static/src/components/configurator/configurator.js

```diff
@@ -617,6 +618,30 @@ async function applyConfigurator(self, themeName) {
         self.env.router.navigate({to: 'CONFIGURATOR_PALETTE_SELECTION_SCREEN'});
         return;
     }
+
+    async function attemptConfiguratorApply(data, retryCount = 0) {
+        try {
+            const resp = await self.rpc({
+                model: 'website',
+                method: 'configurator_apply',
+                kwargs: data,
+            });
+            window.sessionStorage.removeItem(SESSION_STORAGE_ITEM_NAME);
+            window.location = resp.url;
+        } catch (error) {
+            if (retryCount < 3) {
+                await concurrency.delay(5000);
+                return attemptConfiguratorApply(data, retryCount + 1);
+            }
+            // Wait a bit before allowing manual retry.
+            window.setTimeout(() => {
```

Comment:
In both the `if` and the virtual `else` you want to wait 5s. You could call `await concurrency.delay(5000);` outside the `if` and not use `setTimeout`. It's not exactly the same as with the `setTimeout` implementation you free the caller of `applyConfigurator` 5 seconds sooner but in this context, I think it's equivalent (and a bit less code).

EDIT: although I guess you would be re-throwing the error 5 seconds later (ok too I guess). I let you choose :wink: